Update to melodic

Olavhaasie opened the pull request:
<!--
To streamline the process of creating and reviewing pull requests, we have created a template.
It is not required to follow this template perfectly, but we encourage you to think about each header.
Before you publish a pull request think about the definition of done for the issue you are trying to resolve.
-->

<!--
Provide the key for the Jira issue this PR resolves.
-->


## Description
Changed kinetic version to melodic.
<!--
Provide a concise description on the feature/fix your pull request implements.
-->

## Changes
<!--
Provide a list of important changes.

e.g.
* Added tests
* Renamed variable
* Added topic
-->

<!-- Please don't forget to request for reviews -->

Commit:
Update to melodic

Commit:
Fix ros distro in travis config

Commit:
Fix ubuntu distro in build

RutgerVanBeek reviewed (dismissed):
Approve, although more needs to be updated, like the use of SMACH and some launch files have changed names.

Olavhaasie committed:
Remove install script from build

Commit:
Remove sphinx-copybutton dependency

The installation of the python2 package would fail because it was not
able to resolve the 'flit' dependency.

Commit:
Fix external links

Commit:
Add correct install for documentation development
